### PR TITLE
Enhance withdrawal flow with price impact and order details

### PR DESCRIPTION
**Pull request by piotrwitek**
Improve the withdrawal flow by adding price impact calculations and enhancing the display of order information, including slippage and transaction fees. Implemented a new utility for calculating price impact and updated relevant components for better formatting.

## Release Notes

- **New Features**
  - Enhanced transaction handling in the sidebar for withdrawal and deposit components.
  - Improved formatting for financial data in transaction displays.
  - Introduced a new function to calculate price impact based on trade prices.

- **Bug Fixes**
  - Refined error handling in the price fetching logic for the OneInchOracleProvider.

- **Documentation**
  - Updated type definitions for transaction-related components and hooks.

- **Chores**
  - Added new dependencies to support enhanced functionalities across various components.

### CODE REVIEW

File: apps/earn-protocol/components/molecules/SidebarElements/OrderInfoDeposit.tsx [hunk 1]
```diff
@@ -39,27 +39,55 @@ export const OrderInfoDeposit = ({
...
-                {slippage.toString()}
+                {formatPercent(slippage.value, { precision: 2 })}
               </Text>
             </>
           )}
           <Text variant="p3semi">Transaction Fee</Text>
           <Text variant="p3semi" className={orderInfoDepositWithdrawStyles.depositDetailsValue}>
-            {transactionFeeLoading ? 'Loading...' : transactionFee}
+            {transactionFeeLoading
+              ? 'Loading...'
```
Comment (piekczyk) [inline apps/earn-protocol/components/molecules/SidebarElements/OrderInfoDeposit.tsx:87]: I think we can use here SkeletonLine

`<SkeletonLine style={{ display: 'inline-block' }} width={200} height={30} />` (props to be configured)

File: apps/earn-protocol/components/molecules/SidebarElements/OrderInfoDeposit.tsx [hunk 2]
```diff
@@ -39,27 +39,55 @@ export const OrderInfoDeposit = ({
...
+              : transactionFee
+                ? `$${formatFiatBalance(transactionFee)}`
+                : 'N/A'}
```
Comment (piekczyk) [inline apps/earn-protocol/components/molecules/SidebarElements/OrderInfoDeposit.tsx:90]: i think `n/a` looks better form UI/UX pov

File: apps/earn-protocol/components/molecules/SidebarElements/OrderInfoWithdraw.tsx [hunk 3]
```diff
@@ -1,26 +1,96 @@
...
+              <Text variant="p3semi">Slippage</Text>
+              <Text variant="p3semi" className={orderInfoDepositWithdrawStyles.depositDetailsValue}>
+                {formatPercent(slippage.value, { precision: 2 })}
+              </Text>
+            </>
+          )}
+          <Text variant="p3semi">Transaction Fee</Text>
+          <Text variant="p3semi" className={orderInfoDepositWithdrawStyles.depositDetailsValue}>
+            {transactionFeeLoading
+              ? 'Loading...'
```
Comment (piekczyk) [inline apps/earn-protocol/components/molecules/SidebarElements/OrderInfoWithdraw.tsx:87]: same comment here regarding skeleton and n/a

**Review by piekczyk — CHANGES_REQUESTED** [review]

File: apps/earn-protocol/components/molecules/SidebarElements/OrderInfoWithdraw.tsx (hunk 3 above)
Comment (piotrwitek) [reply apps/earn-protocol/components/molecules/SidebarElements/OrderInfoWithdraw.tsx:87]: thanks

**Review by marcinciarka — DISMISSED** [review]

File: apps/earn-protocol/components/molecules/SidebarElements/OrderInfoDeposit.tsx (hunk 2 above)
Comment (piotrwitek) [reply apps/earn-protocol/components/molecules/SidebarElements/OrderInfoDeposit.tsx:90]: done

File: apps/earn-protocol/components/molecules/SidebarElements/OrderInfoWithdraw.tsx (hunk 3 above)
Comment (piotrwitek) [reply apps/earn-protocol/components/molecules/SidebarElements/OrderInfoWithdraw.tsx:87]: done

**Pull request merged**